app/models/game.rb

```diff
@@ -23,6 +24,91 @@ class Game < ActiveRecord::Base
 
   before_save :sync_current_inning_with_game_status, :prevent_mirror_matches
 
+  def next_half_inning
+    completed_half_innings = innings.completed.count
+
+    if innings.in_progress.count > 0
+      innings.in_progress.first
+    else
+      Inning.new(
+        game: self,
+        half: innings.completed.count % 2 == 0 ? :top : :bottom,
+        status: :in_progress,
+        number: next_inning(completed_half_innings),
+        runs: 0  
+      )
+    end
+  end
+
+  def going_final?
```

Comment:
This needs a test and possibly a refactor. Also it appears I had ideas of making it a private method at one time. See if this may still apply.